// @aws-sdk/client-lightsail/dist-cjs/Lightsail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Lightsail = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const AllocateStaticIpCommand_1 = require("./commands/AllocateStaticIpCommand");
const AttachCertificateToDistributionCommand_1 = require("./commands/AttachCertificateToDistributionCommand");
const AttachDiskCommand_1 = require("./commands/AttachDiskCommand");
const AttachInstancesToLoadBalancerCommand_1 = require("./commands/AttachInstancesToLoadBalancerCommand");
const AttachLoadBalancerTlsCertificateCommand_1 = require("./commands/AttachLoadBalancerTlsCertificateCommand");
const AttachStaticIpCommand_1 = require("./commands/AttachStaticIpCommand");
const CloseInstancePublicPortsCommand_1 = require("./commands/CloseInstancePublicPortsCommand");
const CopySnapshotCommand_1 = require("./commands/CopySnapshotCommand");
const CreateBucketAccessKeyCommand_1 = require("./commands/CreateBucketAccessKeyCommand");
const CreateBucketCommand_1 = require("./commands/CreateBucketCommand");
const CreateCertificateCommand_1 = require("./commands/CreateCertificateCommand");
const CreateCloudFormationStackCommand_1 = require("./commands/CreateCloudFormationStackCommand");
const CreateContactMethodCommand_1 = require("./commands/CreateContactMethodCommand");
const CreateContainerServiceCommand_1 = require("./commands/CreateContainerServiceCommand");
const CreateContainerServiceDeploymentCommand_1 = require("./commands/CreateContainerServiceDeploymentCommand");
const CreateContainerServiceRegistryLoginCommand_1 = require("./commands/CreateContainerServiceRegistryLoginCommand");
const CreateDiskCommand_1 = require("./commands/CreateDiskCommand");
const CreateDiskFromSnapshotCommand_1 = require("./commands/CreateDiskFromSnapshotCommand");
const CreateDiskSnapshotCommand_1 = require("./commands/CreateDiskSnapshotCommand");
const CreateDistributionCommand_1 = require("./commands/CreateDistributionCommand");
const CreateDomainCommand_1 = require("./commands/CreateDomainCommand");
const CreateDomainEntryCommand_1 = require("./commands/CreateDomainEntryCommand");
const CreateGUISessionAccessDetailsCommand_1 = require("./commands/CreateGUISessionAccessDetailsCommand");
const CreateInstancesCommand_1 = require("./commands/CreateInstancesCommand");
const CreateInstancesFromSnapshotCommand_1 = require("./commands/CreateInstancesFromSnapshotCommand");
const CreateInstanceSnapshotCommand_1 = require("./commands/CreateInstanceSnapshotCommand");
const CreateKeyPairCommand_1 = require("./commands/CreateKeyPairCommand");
const CreateLoadBalancerCommand_1 = require("./commands/CreateLoadBalancerCommand");
const CreateLoadBalancerTlsCertificateCommand_1 = require("./commands/CreateLoadBalancerTlsCertificateCommand");
const CreateRelationalDatabaseCommand_1 = require("./commands/CreateRelationalDatabaseCommand");
const CreateRelationalDatabaseFromSnapshotCommand_1 = require("./commands/CreateRelationalDatabaseFromSnapshotCommand");
const CreateRelationalDatabaseSnapshotCommand_1 = require("./commands/CreateRelationalDatabaseSnapshotCommand");
const DeleteAlarmCommand_1 = require("./commands/DeleteAlarmCommand");
const DeleteAutoSnapshotCommand_1 = require("./commands/DeleteAutoSnapshotCommand");
const DeleteBucketAccessKeyCommand_1 = require("./commands/DeleteBucketAccessKeyCommand");
const DeleteBucketCommand_1 = require("./commands/DeleteBucketCommand");
const DeleteCertificateCommand_1 = require("./commands/DeleteCertificateCommand");
const DeleteContactMethodCommand_1 = require("./commands/DeleteContactMethodCommand");
const DeleteContainerImageCommand_1 = require("./commands/DeleteContainerImageCommand");
const DeleteContainerServiceCommand_1 = require("./commands/DeleteContainerServiceCommand");
const DeleteDiskCommand_1 = require("./commands/DeleteDiskCommand");
const DeleteDiskSnapshotCommand_1 = require("./commands/DeleteDiskSnapshotCommand");
const DeleteDistributionCommand_1 = require("./commands/DeleteDistributionCommand");
const DeleteDomainCommand_1 = require("./commands/DeleteDomainCommand");
const DeleteDomainEntryCommand_1 = require("./commands/DeleteDomainEntryCommand");
const DeleteInstanceCommand_1 = require("./commands/DeleteInstanceCommand");
const DeleteInstanceSnapshotCommand_1 = require("./commands/DeleteInstanceSnapshotCommand");
const DeleteKeyPairCommand_1 = require("./commands/DeleteKeyPairCommand");
const DeleteKnownHostKeysCommand_1 = require("./commands/DeleteKnownHostKeysCommand");
const DeleteLoadBalancerCommand_1 = require("./commands/DeleteLoadBalancerCommand");
const DeleteLoadBalancerTlsCertificateCommand_1 = require("./commands/DeleteLoadBalancerTlsCertificateCommand");
const DeleteRelationalDatabaseCommand_1 = require("./commands/DeleteRelationalDatabaseCommand");
const DeleteRelationalDatabaseSnapshotCommand_1 = require("./commands/DeleteRelationalDatabaseSnapshotCommand");
const DetachCertificateFromDistributionCommand_1 = require("./commands/DetachCertificateFromDistributionCommand");
const DetachDiskCommand_1 = require("./commands/DetachDiskCommand");
const DetachInstancesFromLoadBalancerCommand_1 = require("./commands/DetachInstancesFromLoadBalancerCommand");
const DetachStaticIpCommand_1 = require("./commands/DetachStaticIpCommand");
const DisableAddOnCommand_1 = require("./commands/DisableAddOnCommand");
const DownloadDefaultKeyPairCommand_1 = require("./commands/DownloadDefaultKeyPairCommand");
const EnableAddOnCommand_1 = require("./commands/EnableAddOnCommand");
const ExportSnapshotCommand_1 = require("./commands/ExportSnapshotCommand");
const GetActiveNamesCommand_1 = require("./commands/GetActiveNamesCommand");
const GetAlarmsCommand_1 = require("./commands/GetAlarmsCommand");
const GetAutoSnapshotsCommand_1 = require("./commands/GetAutoSnapshotsCommand");
const GetBlueprintsCommand_1 = require("./commands/GetBlueprintsCommand");
const GetBucketAccessKeysCommand_1 = require("./commands/GetBucketAccessKeysCommand");
const GetBucketBundlesCommand_1 = require("./commands/GetBucketBundlesCommand");
const GetBucketMetricDataCommand_1 = require("./commands/GetBucketMetricDataCommand");
const GetBucketsCommand_1 = require("./commands/GetBucketsCommand");
const GetBundlesCommand_1 = require("./commands/GetBundlesCommand");
const GetCertificatesCommand_1 = require("./commands/GetCertificatesCommand");
const GetCloudFormationStackRecordsCommand_1 = require("./commands/GetCloudFormationStackRecordsCommand");
const GetContactMethodsCommand_1 = require("./commands/GetContactMethodsCommand");
const GetContainerAPIMetadataCommand_1 = require("./commands/GetContainerAPIMetadataCommand");
const GetContainerImagesCommand_1 = require("./commands/GetContainerImagesCommand");
const GetContainerLogCommand_1 = require("./commands/GetContainerLogCommand");
const GetContainerServiceDeploymentsCommand_1 = require("./commands/GetContainerServiceDeploymentsCommand");
const GetContainerServiceMetricDataCommand_1 = require("./commands/GetContainerServiceMetricDataCommand");
const GetContainerServicePowersCommand_1 = require("./commands/GetContainerServicePowersCommand");
const GetContainerServicesCommand_1 = require("./commands/GetContainerServicesCommand");
const GetCostEstimateCommand_1 = require("./commands/GetCostEstimateCommand");
const GetDiskCommand_1 = require("./commands/GetDiskCommand");
const GetDisksCommand_1 = require("./commands/GetDisksCommand");
const GetDiskSnapshotCommand_1 = require("./commands/GetDiskSnapshotCommand");
const GetDiskSnapshotsCommand_1 = require("./commands/GetDiskSnapshotsCommand");
const GetDistributionBundlesCommand_1 = require("./commands/GetDistributionBundlesCommand");
const GetDistributionLatestCacheResetCommand_1 = require("./commands/GetDistributionLatestCacheResetCommand");
const GetDistributionMetricDataCommand_1 = require("./commands/GetDistributionMetricDataCommand");
const GetDistributionsCommand_1 = require("./commands/GetDistributionsCommand");
const GetDomainCommand_1 = require("./commands/GetDomainCommand");
const GetDomainsCommand_1 = require("./commands/GetDomainsCommand");
const GetExportSnapshotRecordsCommand_1 = require("./commands/GetExportSnapshotRecordsCommand");
const GetInstanceAccessDetailsCommand_1 = require("./commands/GetInstanceAccessDetailsCommand");
const GetInstanceCommand_1 = require("./commands/GetInstanceCommand");
const GetInstanceMetricDataCommand_1 = require("./commands/GetInstanceMetricDataCommand");
const GetInstancePortStatesCommand_1 = require("./commands/GetInstancePortStatesCommand");
const GetInstancesCommand_1 = require("./commands/GetInstancesCommand");
const GetInstanceSnapshotCommand_1 = require("./commands/GetInstanceSnapshotCommand");
const GetInstanceSnapshotsCommand_1 = require("./commands/GetInstanceSnapshotsCommand");
const GetInstanceStateCommand_1 = require("./commands/GetInstanceStateCommand");
const GetKeyPairCommand_1 = require("./commands/GetKeyPairCommand");
const GetKeyPairsCommand_1 = require("./commands/GetKeyPairsCommand");
const GetLoadBalancerCommand_1 = require("./commands/GetLoadBalancerCommand");
const GetLoadBalancerMetricDataCommand_1 = require("./commands/GetLoadBalancerMetricDataCommand");
const GetLoadBalancersCommand_1 = require("./commands/GetLoadBalancersCommand");
const GetLoadBalancerTlsCertificatesCommand_1 = require("./commands/GetLoadBalancerTlsCertificatesCommand");
const GetLoadBalancerTlsPoliciesCommand_1 = require("./commands/GetLoadBalancerTlsPoliciesCommand");
const GetOperationCommand_1 = require("./commands/GetOperationCommand");
const GetOperationsCommand_1 = require("./commands/GetOperationsCommand");
const GetOperationsForResourceCommand_1 = require("./commands/GetOperationsForResourceCommand");
const GetRegionsCommand_1 = require("./commands/GetRegionsCommand");
const GetRelationalDatabaseBlueprintsCommand_1 = require("./commands/GetRelationalDatabaseBlueprintsCommand");
const GetRelationalDatabaseBundlesCommand_1 = require("./commands/GetRelationalDatabaseBundlesCommand");
const GetRelationalDatabaseCommand_1 = require("./commands/GetRelationalDatabaseCommand");
const GetRelationalDatabaseEventsCommand_1 = require("./commands/GetRelationalDatabaseEventsCommand");
const GetRelationalDatabaseLogEventsCommand_1 = require("./commands/GetRelationalDatabaseLogEventsCommand");
const GetRelationalDatabaseLogStreamsCommand_1 = require("./commands/GetRelationalDatabaseLogStreamsCommand");
const GetRelationalDatabaseMasterUserPasswordCommand_1 = require("./commands/GetRelationalDatabaseMasterUserPasswordCommand");
const GetRelationalDatabaseMetricDataCommand_1 = require("./commands/GetRelationalDatabaseMetricDataCommand");
const GetRelationalDatabaseParametersCommand_1 = require("./commands/GetRelationalDatabaseParametersCommand");
const GetRelationalDatabasesCommand_1 = require("./commands/GetRelationalDatabasesCommand");
const GetRelationalDatabaseSnapshotCommand_1 = require("./commands/GetRelationalDatabaseSnapshotCommand");
const GetRelationalDatabaseSnapshotsCommand_1 = require("./commands/GetRelationalDatabaseSnapshotsCommand");
const GetStaticIpCommand_1 = require("./commands/GetStaticIpCommand");
const GetStaticIpsCommand_1 = require("./commands/GetStaticIpsCommand");
const ImportKeyPairCommand_1 = require("./commands/ImportKeyPairCommand");
const IsVpcPeeredCommand_1 = require("./commands/IsVpcPeeredCommand");
const OpenInstancePublicPortsCommand_1 = require("./commands/OpenInstancePublicPortsCommand");
const PeerVpcCommand_1 = require("./commands/PeerVpcCommand");
const PutAlarmCommand_1 = require("./commands/PutAlarmCommand");
const PutInstancePublicPortsCommand_1 = require("./commands/PutInstancePublicPortsCommand");
const RebootInstanceCommand_1 = require("./commands/RebootInstanceCommand");
const RebootRelationalDatabaseCommand_1 = require("./commands/RebootRelationalDatabaseCommand");
const RegisterContainerImageCommand_1 = require("./commands/RegisterContainerImageCommand");
const ReleaseStaticIpCommand_1 = require("./commands/ReleaseStaticIpCommand");
const ResetDistributionCacheCommand_1 = require("./commands/ResetDistributionCacheCommand");
const SendContactMethodVerificationCommand_1 = require("./commands/SendContactMethodVerificationCommand");
const SetIpAddressTypeCommand_1 = require("./commands/SetIpAddressTypeCommand");
const SetResourceAccessForBucketCommand_1 = require("./commands/SetResourceAccessForBucketCommand");
const StartGUISessionCommand_1 = require("./commands/StartGUISessionCommand");
const StartInstanceCommand_1 = require("./commands/StartInstanceCommand");
const StartRelationalDatabaseCommand_1 = require("./commands/StartRelationalDatabaseCommand");
const StopGUISessionCommand_1 = require("./commands/StopGUISessionCommand");
const StopInstanceCommand_1 = require("./commands/StopInstanceCommand");
const StopRelationalDatabaseCommand_1 = require("./commands/StopRelationalDatabaseCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const TestAlarmCommand_1 = require("./commands/TestAlarmCommand");
const UnpeerVpcCommand_1 = require("./commands/UnpeerVpcCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateBucketBundleCommand_1 = require("./commands/UpdateBucketBundleCommand");
const UpdateBucketCommand_1 = require("./commands/UpdateBucketCommand");
const UpdateContainerServiceCommand_1 = require("./commands/UpdateContainerServiceCommand");
const UpdateDistributionBundleCommand_1 = require("./commands/UpdateDistributionBundleCommand");
const UpdateDistributionCommand_1 = require("./commands/UpdateDistributionCommand");
const UpdateDomainEntryCommand_1 = require("./commands/UpdateDomainEntryCommand");
const UpdateInstanceMetadataOptionsCommand_1 = require("./commands/UpdateInstanceMetadataOptionsCommand");
const UpdateLoadBalancerAttributeCommand_1 = require("./commands/UpdateLoadBalancerAttributeCommand");
const UpdateRelationalDatabaseCommand_1 = require("./commands/UpdateRelationalDatabaseCommand");
const UpdateRelationalDatabaseParametersCommand_1 = require("./commands/UpdateRelationalDatabaseParametersCommand");
const LightsailClient_1 = require("./LightsailClient");
const commands = {
    AllocateStaticIpCommand: AllocateStaticIpCommand_1.AllocateStaticIpCommand,
    AttachCertificateToDistributionCommand: AttachCertificateToDistributionCommand_1.AttachCertificateToDistributionCommand,
    AttachDiskCommand: AttachDiskCommand_1.AttachDiskCommand,
    AttachInstancesToLoadBalancerCommand: AttachInstancesToLoadBalancerCommand_1.AttachInstancesToLoadBalancerCommand,
    AttachLoadBalancerTlsCertificateCommand: AttachLoadBalancerTlsCertificateCommand_1.AttachLoadBalancerTlsCertificateCommand,
    AttachStaticIpCommand: AttachStaticIpCommand_1.AttachStaticIpCommand,
    CloseInstancePublicPortsCommand: CloseInstancePublicPortsCommand_1.CloseInstancePublicPortsCommand,
    CopySnapshotCommand: CopySnapshotCommand_1.CopySnapshotCommand,
    CreateBucketCommand: CreateBucketCommand_1.CreateBucketCommand,
    CreateBucketAccessKeyCommand: CreateBucketAccessKeyCommand_1.CreateBucketAccessKeyCommand,
    CreateCertificateCommand: CreateCertificateCommand_1.CreateCertificateCommand,
    CreateCloudFormationStackCommand: CreateCloudFormationStackCommand_1.CreateCloudFormationStackCommand,
    CreateContactMethodCommand: CreateContactMethodCommand_1.CreateContactMethodCommand,
    CreateContainerServiceCommand: CreateContainerServiceCommand_1.CreateContainerServiceCommand,
    CreateContainerServiceDeploymentCommand: CreateContainerServiceDeploymentCommand_1.CreateContainerServiceDeploymentCommand,
    CreateContainerServiceRegistryLoginCommand: CreateContainerServiceRegistryLoginCommand_1.CreateContainerServiceRegistryLoginCommand,
    CreateDiskCommand: CreateDiskCommand_1.CreateDiskCommand,
    CreateDiskFromSnapshotCommand: CreateDiskFromSnapshotCommand_1.CreateDiskFromSnapshotCommand,
    CreateDiskSnapshotCommand: CreateDiskSnapshotCommand_1.CreateDiskSnapshotCommand,
    CreateDistributionCommand: CreateDistributionCommand_1.CreateDistributionCommand,
    CreateDomainCommand: CreateDomainCommand_1.CreateDomainCommand,
    CreateDomainEntryCommand: CreateDomainEntryCommand_1.CreateDomainEntryCommand,
    CreateGUISessionAccessDetailsCommand: CreateGUISessionAccessDetailsCommand_1.CreateGUISessionAccessDetailsCommand,
    CreateInstancesCommand: CreateInstancesCommand_1.CreateInstancesCommand,
    CreateInstancesFromSnapshotCommand: CreateInstancesFromSnapshotCommand_1.CreateInstancesFromSnapshotCommand,
    CreateInstanceSnapshotCommand: CreateInstanceSnapshotCommand_1.CreateInstanceSnapshotCommand,
    CreateKeyPairCommand: CreateKeyPairCommand_1.CreateKeyPairCommand,
    CreateLoadBalancerCommand: CreateLoadBalancerCommand_1.CreateLoadBalancerCommand,
    CreateLoadBalancerTlsCertificateCommand: CreateLoadBalancerTlsCertificateCommand_1.CreateLoadBalancerTlsCertificateCommand,
    CreateRelationalDatabaseCommand: CreateRelationalDatabaseCommand_1.CreateRelationalDatabaseCommand,
    CreateRelationalDatabaseFromSnapshotCommand: CreateRelationalDatabaseFromSnapshotCommand_1.CreateRelationalDatabaseFromSnapshotCommand,
    CreateRelationalDatabaseSnapshotCommand: CreateRelationalDatabaseSnapshotCommand_1.CreateRelationalDatabaseSnapshotCommand,
    DeleteAlarmCommand: DeleteAlarmCommand_1.DeleteAlarmCommand,
    DeleteAutoSnapshotCommand: DeleteAutoSnapshotCommand_1.DeleteAutoSnapshotCommand,
    DeleteBucketCommand: DeleteBucketCommand_1.DeleteBucketCommand,
    DeleteBucketAccessKeyCommand: DeleteBucketAccessKeyCommand_1.DeleteBucketAccessKeyCommand,
    DeleteCertificateCommand: DeleteCertificateCommand_1.DeleteCertificateCommand,
    DeleteContactMethodCommand: DeleteContactMethodCommand_1.DeleteContactMethodCommand,
    DeleteContainerImageCommand: DeleteContainerImageCommand_1.DeleteContainerImageCommand,
    DeleteContainerServiceCommand: DeleteContainerServiceCommand_1.DeleteContainerServiceCommand,
    DeleteDiskCommand: DeleteDiskCommand_1.DeleteDiskCommand,
    DeleteDiskSnapshotCommand: DeleteDiskSnapshotCommand_1.DeleteDiskSnapshotCommand,
    DeleteDistributionCommand: DeleteDistributionCommand_1.DeleteDistributionCommand,
    DeleteDomainCommand: DeleteDomainCommand_1.DeleteDomainCommand,
    DeleteDomainEntryCommand: DeleteDomainEntryCommand_1.DeleteDomainEntryCommand,
    DeleteInstanceCommand: DeleteInstanceCommand_1.DeleteInstanceCommand,
    DeleteInstanceSnapshotCommand: DeleteInstanceSnapshotCommand_1.DeleteInstanceSnapshotCommand,
    DeleteKeyPairCommand: DeleteKeyPairCommand_1.DeleteKeyPairCommand,
    DeleteKnownHostKeysCommand: DeleteKnownHostKeysCommand_1.DeleteKnownHostKeysCommand,
    DeleteLoadBalancerCommand: DeleteLoadBalancerCommand_1.DeleteLoadBalancerCommand,
    DeleteLoadBalancerTlsCertificateCommand: DeleteLoadBalancerTlsCertificateCommand_1.DeleteLoadBalancerTlsCertificateCommand,
    DeleteRelationalDatabaseCommand: DeleteRelationalDatabaseCommand_1.DeleteRelationalDatabaseCommand,
    DeleteRelationalDatabaseSnapshotCommand: DeleteRelationalDatabaseSnapshotCommand_1.DeleteRelationalDatabaseSnapshotCommand,
    DetachCertificateFromDistributionCommand: DetachCertificateFromDistributionCommand_1.DetachCertificateFromDistributionCommand,
    DetachDiskCommand: DetachDiskCommand_1.DetachDiskCommand,
    DetachInstancesFromLoadBalancerCommand: DetachInstancesFromLoadBalancerCommand_1.DetachInstancesFromLoadBalancerCommand,
    DetachStaticIpCommand: DetachStaticIpCommand_1.DetachStaticIpCommand,
    DisableAddOnCommand: DisableAddOnCommand_1.DisableAddOnCommand,
    DownloadDefaultKeyPairCommand: DownloadDefaultKeyPairCommand_1.DownloadDefaultKeyPairCommand,
    EnableAddOnCommand: EnableAddOnCommand_1.EnableAddOnCommand,
    ExportSnapshotCommand: ExportSnapshotCommand_1.ExportSnapshotCommand,
    GetActiveNamesCommand: GetActiveNamesCommand_1.GetActiveNamesCommand,
    GetAlarmsCommand: GetAlarmsCommand_1.GetAlarmsCommand,
    GetAutoSnapshotsCommand: GetAutoSnapshotsCommand_1.GetAutoSnapshotsCommand,
    GetBlueprintsCommand: GetBlueprintsCommand_1.GetBlueprintsCommand,
    GetBucketAccessKeysCommand: GetBucketAccessKeysCommand_1.GetBucketAccessKeysCommand,
    GetBucketBundlesCommand: GetBucketBundlesCommand_1.GetBucketBundlesCommand,
    GetBucketMetricDataCommand: GetBucketMetricDataCommand_1.GetBucketMetricDataCommand,
    GetBucketsCommand: GetBucketsCommand_1.GetBucketsCommand,
    GetBundlesCommand: GetBundlesCommand_1.GetBundlesCommand,
    GetCertificatesCommand: GetCertificatesCommand_1.GetCertificatesCommand,
    GetCloudFormationStackRecordsCommand: GetCloudFormationStackRecordsCommand_1.GetCloudFormationStackRecordsCommand,
    GetContactMethodsCommand: GetContactMethodsCommand_1.GetContactMethodsCommand,
    GetContainerAPIMetadataCommand: GetContainerAPIMetadataCommand_1.GetContainerAPIMetadataCommand,
    GetContainerImagesCommand: GetContainerImagesCommand_1.GetContainerImagesCommand,
    GetContainerLogCommand: GetContainerLogCommand_1.GetContainerLogCommand,
    GetContainerServiceDeploymentsCommand: GetContainerServiceDeploymentsCommand_1.GetContainerServiceDeploymentsCommand,
    GetContainerServiceMetricDataCommand: GetContainerServiceMetricDataCommand_1.GetContainerServiceMetricDataCommand,
    GetContainerServicePowersCommand: GetContainerServicePowersCommand_1.GetContainerServicePowersCommand,
    GetContainerServicesCommand: GetContainerServicesCommand_1.GetContainerServicesCommand,
    GetCostEstimateCommand: GetCostEstimateCommand_1.GetCostEstimateCommand,
    GetDiskCommand: GetDiskCommand_1.GetDiskCommand,
    GetDisksCommand: GetDisksCommand_1.GetDisksCommand,
    GetDiskSnapshotCommand: GetDiskSnapshotCommand_1.GetDiskSnapshotCommand,
    GetDiskSnapshotsCommand: GetDiskSnapshotsCommand_1.GetDiskSnapshotsCommand,
    GetDistributionBundlesCommand: GetDistributionBundlesCommand_1.GetDistributionBundlesCommand,
    GetDistributionLatestCacheResetCommand: GetDistributionLatestCacheResetCommand_1.GetDistributionLatestCacheResetCommand,
    GetDistributionMetricDataCommand: GetDistributionMetricDataCommand_1.GetDistributionMetricDataCommand,
    GetDistributionsCommand: GetDistributionsCommand_1.GetDistributionsCommand,
    GetDomainCommand: GetDomainCommand_1.GetDomainCommand,
    GetDomainsCommand: GetDomainsCommand_1.GetDomainsCommand,
    GetExportSnapshotRecordsCommand: GetExportSnapshotRecordsCommand_1.GetExportSnapshotRecordsCommand,
    GetInstanceCommand: GetInstanceCommand_1.GetInstanceCommand,
    GetInstanceAccessDetailsCommand: GetInstanceAccessDetailsCommand_1.GetInstanceAccessDetailsCommand,
    GetInstanceMetricDataCommand: GetInstanceMetricDataCommand_1.GetInstanceMetricDataCommand,
    GetInstancePortStatesCommand: GetInstancePortStatesCommand_1.GetInstancePortStatesCommand,
    GetInstancesCommand: GetInstancesCommand_1.GetInstancesCommand,
    GetInstanceSnapshotCommand: GetInstanceSnapshotCommand_1.GetInstanceSnapshotCommand,
    GetInstanceSnapshotsCommand: GetInstanceSnapshotsCommand_1.GetInstanceSnapshotsCommand,
    GetInstanceStateCommand: GetInstanceStateCommand_1.GetInstanceStateCommand,
    GetKeyPairCommand: GetKeyPairCommand_1.GetKeyPairCommand,
    GetKeyPairsCommand: GetKeyPairsCommand_1.GetKeyPairsCommand,
    GetLoadBalancerCommand: GetLoadBalancerCommand_1.GetLoadBalancerCommand,
    GetLoadBalancerMetricDataCommand: GetLoadBalancerMetricDataCommand_1.GetLoadBalancerMetricDataCommand,
    GetLoadBalancersCommand: GetLoadBalancersCommand_1.GetLoadBalancersCommand,
    GetLoadBalancerTlsCertificatesCommand: GetLoadBalancerTlsCertificatesCommand_1.GetLoadBalancerTlsCertificatesCommand,
    GetLoadBalancerTlsPoliciesCommand: GetLoadBalancerTlsPoliciesCommand_1.GetLoadBalancerTlsPoliciesCommand,
    GetOperationCommand: GetOperationCommand_1.GetOperationCommand,
    GetOperationsCommand: GetOperationsCommand_1.GetOperationsCommand,
    GetOperationsForResourceCommand: GetOperationsForResourceCommand_1.GetOperationsForResourceCommand,
    GetRegionsCommand: GetRegionsCommand_1.GetRegionsCommand,
    GetRelationalDatabaseCommand: GetRelationalDatabaseCommand_1.GetRelationalDatabaseCommand,
    GetRelationalDatabaseBlueprintsCommand: GetRelationalDatabaseBlueprintsCommand_1.GetRelationalDatabaseBlueprintsCommand,
    GetRelationalDatabaseBundlesCommand: GetRelationalDatabaseBundlesCommand_1.GetRelationalDatabaseBundlesCommand,
    GetRelationalDatabaseEventsCommand: GetRelationalDatabaseEventsCommand_1.GetRelationalDatabaseEventsCommand,
    GetRelationalDatabaseLogEventsCommand: GetRelationalDatabaseLogEventsCommand_1.GetRelationalDatabaseLogEventsCommand,
    GetRelationalDatabaseLogStreamsCommand: GetRelationalDatabaseLogStreamsCommand_1.GetRelationalDatabaseLogStreamsCommand,
    GetRelationalDatabaseMasterUserPasswordCommand: GetRelationalDatabaseMasterUserPasswordCommand_1.GetRelationalDatabaseMasterUserPasswordCommand,
    GetRelationalDatabaseMetricDataCommand: GetRelationalDatabaseMetricDataCommand_1.GetRelationalDatabaseMetricDataCommand,
    GetRelationalDatabaseParametersCommand: GetRelationalDatabaseParametersCommand_1.GetRelationalDatabaseParametersCommand,
    GetRelationalDatabasesCommand: GetRelationalDatabasesCommand_1.GetRelationalDatabasesCommand,
    GetRelationalDatabaseSnapshotCommand: GetRelationalDatabaseSnapshotCommand_1.GetRelationalDatabaseSnapshotCommand,
    GetRelationalDatabaseSnapshotsCommand: GetRelationalDatabaseSnapshotsCommand_1.GetRelationalDatabaseSnapshotsCommand,
    GetStaticIpCommand: GetStaticIpCommand_1.GetStaticIpCommand,
    GetStaticIpsCommand: GetStaticIpsCommand_1.GetStaticIpsCommand,
    ImportKeyPairCommand: ImportKeyPairCommand_1.ImportKeyPairCommand,
    IsVpcPeeredCommand: IsVpcPeeredCommand_1.IsVpcPeeredCommand,
    OpenInstancePublicPortsCommand: OpenInstancePublicPortsCommand_1.OpenInstancePublicPortsCommand,
    PeerVpcCommand: PeerVpcCommand_1.PeerVpcCommand,
    PutAlarmCommand: PutAlarmCommand_1.PutAlarmCommand,
    PutInstancePublicPortsCommand: PutInstancePublicPortsCommand_1.PutInstancePublicPortsCommand,
    RebootInstanceCommand: RebootInstanceCommand_1.RebootInstanceCommand,
    RebootRelationalDatabaseCommand: RebootRelationalDatabaseCommand_1.RebootRelationalDatabaseCommand,
    RegisterContainerImageCommand: RegisterContainerImageCommand_1.RegisterContainerImageCommand,
    ReleaseStaticIpCommand: ReleaseStaticIpCommand_1.ReleaseStaticIpCommand,
    ResetDistributionCacheCommand: ResetDistributionCacheCommand_1.ResetDistributionCacheCommand,
    SendContactMethodVerificationCommand: SendContactMethodVerificationCommand_1.SendContactMethodVerificationCommand,
    SetIpAddressTypeCommand: SetIpAddressTypeCommand_1.SetIpAddressTypeCommand,
    SetResourceAccessForBucketCommand: SetResourceAccessForBucketCommand_1.SetResourceAccessForBucketCommand,
    StartGUISessionCommand: StartGUISessionCommand_1.StartGUISessionCommand,
    StartInstanceCommand: StartInstanceCommand_1.StartInstanceCommand,
    StartRelationalDatabaseCommand: StartRelationalDatabaseCommand_1.StartRelationalDatabaseCommand,
    StopGUISessionCommand: StopGUISessionCommand_1.StopGUISessionCommand,
    StopInstanceCommand: StopInstanceCommand_1.StopInstanceCommand,
    StopRelationalDatabaseCommand: StopRelationalDatabaseCommand_1.StopRelationalDatabaseCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    TestAlarmCommand: TestAlarmCommand_1.TestAlarmCommand,
    UnpeerVpcCommand: UnpeerVpcCommand_1.UnpeerVpcCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateBucketCommand: UpdateBucketCommand_1.UpdateBucketCommand,
    UpdateBucketBundleCommand: UpdateBucketBundleCommand_1.UpdateBucketBundleCommand,
    UpdateContainerServiceCommand: UpdateContainerServiceCommand_1.UpdateContainerServiceCommand,
    UpdateDistributionCommand: UpdateDistributionCommand_1.UpdateDistributionCommand,
    UpdateDistributionBundleCommand: UpdateDistributionBundleCommand_1.UpdateDistributionBundleCommand,
    UpdateDomainEntryCommand: UpdateDomainEntryCommand_1.UpdateDomainEntryCommand,
    UpdateInstanceMetadataOptionsCommand: UpdateInstanceMetadataOptionsCommand_1.UpdateInstanceMetadataOptionsCommand,
    UpdateLoadBalancerAttributeCommand: UpdateLoadBalancerAttributeCommand_1.UpdateLoadBalancerAttributeCommand,
    UpdateRelationalDatabaseCommand: UpdateRelationalDatabaseCommand_1.UpdateRelationalDatabaseCommand,
    UpdateRelationalDatabaseParametersCommand: UpdateRelationalDatabaseParametersCommand_1.UpdateRelationalDatabaseParametersCommand,
};
class Lightsail extends LightsailClient_1.LightsailClient {
}
exports.Lightsail = Lightsail;
(0, smithy_client_1.createAggregatedClient)(commands, Lightsail);
